src/Microsoft.AspNetCore.Razor.Language/Legacy/Block.cs

```diff
@@ -11,6 +11,8 @@ namespace Microsoft.AspNetCore.Razor.Language.Legacy
 {
     internal class Block : SyntaxTreeNode
     {
+        private int _length = -1;
```

Comment:
use `int?` it does the same thing but it's much more semantically meaningful

Reply:
Is memory consumption a concern?

Reply:
Not that we've seen.